src/App.js

```diff
@@ -36,22 +36,22 @@ export default function ActivityApp() {
   return (
     <Appcontainer>
       {isHidden === false && (
-        <FavoritesPage activities={activities} favorite={favorite} isHidden={isHidden} onSetIsHidden={setIsHidden} />
-      )}
-      {isHidden === true && (
         <DurationFilter onFilterDurationReset={filterDurationReset} onFilterDurationValue={setDurationFilterValue} />
       )}
-      {isHidden === true && <TypeFilter onFilterTypeReset={filterTypeReset} onFilterTypeValue={setTypeFiltervalue} />}
-      {isHidden === true && (
+      {isHidden === false && <TypeFilter onFilterTypeReset={filterTypeReset} onFilterTypeValue={setTypeFiltervalue} />}
+      {isHidden === false && (
         <ActivityCards
-          activities={filteredAll}
+          activities={filteredActivities}
           onSetActivities={setActivities}
           favorite={favorite}
           onSetFavorite={setFavorite}
           isHidden={isHidden}
           onSetIsHidden={setIsHidden}
         />
       )}
+      {isHidden && (
+        <FavoritesPage activities={activities} favorite={favorite} isHidden={isHidden} onSetIsHidden={setIsHidden} />
```

Comment:
For example with `onSetIsHidden`: In your `FavoritesPage` you update the state by passing the current state `isHidden` and do:
```js
onClick={() => onSetIsHidden(!isHidden)}
```
This should be in the `App.js` instead:
```suggestion
<FavoritesPage activities={activities} favorite={favorite} onSetIsHidden={() => onSetIsHidden(!isHidden)} />
```
And in your `FavoritesPage`  you would just refer to the callback function:
```js
export default function FavoritesPage({activities, favorite, onSetIsHidden}) {
  //----- Find favorites for Mapping -----
  const showFavorite = activities.filter(activity => favorite.includes(activity.id));

  return (
    <div>
      <h2>Favorites</h2>
      <button onClick={onSetIsHidden}>go back</button>
```
This way you don't need to add the `hidden` state to the component and it prevents from bugs which might happen when you update the state within your page and not where the state is (app.js)

Reply:
Thank you! 

> This way you don't need to add the hidden state to the component and it prevents from bugs which might happen when you update the state within your page and not where the state is (app.js)

That's the part I was kind of missing.